app/forms/idv/api_image_upload_form.rb

```diff
@@ -348,6 +348,7 @@ def update_analytics(client_response:, vendor_request_time_in_ms:)
       update_funnel(client_response)
       birth_year = client_response.pii_from_doc&.dob&.to_date&.year
       zip_code = client_response.pii_from_doc&.zipcode&.to_s&.strip&.slice(0, 5)
+      issue_year = client_response.pii_from_doc&.state_id_issued&.slice(0, 4)
```

Comment:
Are we confident this will be a string? It looks like [TrueID parses this into a `Date`](https://github.com/18F/identity-idp/blob/b956986605a73615cc8cc61dd0fd2e647c5f9bcd/app/services/doc_auth/lexis_nexis/doc_pii_reader.rb#L49-L53), but [`MOCK_IDV_APPLICANT` uses a string](https://github.com/18F/identity-idp/blob/b956986605a73615cc8cc61dd0fd2e647c5f9bcd/lib/idp/constants.rb#L104).

In any case, i feel like we should properly parse and read the `year` from a date here rather than string slicing

Reply:
+1 maybe we could use `to_date.year` (with nil-safe, like above) which would convert strings into dates if needed

Reply:
Tried updating the mock client to deal in Dates, not Strings, since that's a more faithful representation of reality. The follow-on effects aren't bad; see what you think. I _can_ back it out and put the solution suggested by Zach, if that's really what we want, but I think this is better.

Reply:
Updating feature specs.

Reply:
OK, the feature specs turned out to be a mudball to untangle for this. Calling it, implemented the solution suggested by Zach. We do want to fix the mock proofer, but this is not the ticket.